### ISSUE TITLE
Feature to relax all querqy FILTER rules via automatic addition of an "OR" filter query

### ISSUE THREAD

**Issue by Paul-Blanchaert** [querqy/querqy#306]
My customer uses "tiles" within his "products" solr index. These tiles pop-up in the SERP when search requests match the keywords in the tile document. These tiles redirect the user to specific marketing pages related to the tile (indeed, a bit like DECORATE querqy rules; but driven by keyword search).
E.g. tiles for a certain brand: when the brand is used in the search request, the tile appears in the SERP amongst the matching items. 
When implementing querqy filter rules that select on the item's field, the tiles are filtered out as tiles don't have this brand field. A workaround is to set the querqy filter rule to explicitly include the tiles regardless of the item brand-field selection (such as "FILTER * (brand_s:XYZ OR type_s:tile)").
As this applies to all querqy filter rules, we want to avoid to have to define this relaxing filter ("OR type_s:tile") rules for every querqy filter rule.
Basically, we don't want to have this "generic" setup to be included in all the querqy filter rules.
For this purpose, I've created a "QuerqyRelaxFilterQueryComponent" that appends this "OR clause" automatically to the generated filter queries by querqy. The setup of this relax fq is done via the config of the QueryComponent:
    <searchComponent name="query" class="...QuerqyRelaxFilterQueryComponent">
        <str name="fq">type:tile</str>
    </searchComponent>
I'm creating this issue and PR in case you might be interested in incorporating this code into the base querqy solution.
